src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/service/AdjudicationMappingService.kt

```diff
@@ -0,0 +1,96 @@
+package uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.service
+
+import com.microsoft.applicationinsights.TelemetryClient
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.flow.toList
+import org.slf4j.LoggerFactory
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.PageImpl
+import org.springframework.data.domain.Pageable
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.config.DuplicateMappingException
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.data.AdjudicationMappingDto
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.jpa.AdjudicationMapping
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.jpa.AdjudicationMappingType
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.jpa.repository.AdjudicationMappingRepository
+
+@Service
+@Transactional(readOnly = true)
+class AdjudicationMappingService(
+  private val adjudicationMappingRepository: AdjudicationMappingRepository,
+  private val telemetryClient: TelemetryClient,
+) {
+  private companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  @Transactional
+  suspend fun createMapping(createMappingRequest: AdjudicationMappingDto) =
```

Comment:
Adjudication "mappings" are really just a record of a migration or a DPS creation, there are no IDs to map as the nomis ID exists in the adjudication service.
The mappings will probably evolve when lower level entities require mapping